src/main/kotlin/com/sipios/springsearch/SpecificationImpl.kt

```diff
@@ -1,40 +1,42 @@
 package com.sipios.springsearch
 
-import org.springframework.data.jpa.domain.Specification
-
+import com.sipios.springsearch.strategies.ParsingStrategy
+import java.util.ArrayList
 import javax.persistence.criteria.CriteriaBuilder
 import javax.persistence.criteria.CriteriaQuery
 import javax.persistence.criteria.Path
 import javax.persistence.criteria.Predicate
 import javax.persistence.criteria.Root
-import java.util.ArrayList
+import org.springframework.data.jpa.domain.Specification
+import org.springframework.http.HttpStatus
+import org.springframework.web.server.ResponseStatusException
 
 /**
  * Implementation of the JPA Specification based on a Search Criteria
  *
  * @see Specification
  *
- * @param <T> THe class on which the specification will be applied
-</T> */
+ * @param <T>The class on which the specification will be applied</T>
+ * */
 class SpecificationImpl<T>(private val criteria: SearchCriteria) : Specification<T> {
+    @Throws(ResponseStatusException::class)
     override fun toPredicate(root: Root<T>, query: CriteriaQuery<*>, builder: CriteriaBuilder): Predicate? {
         val nestedKey = criteria.key.split(".")
         val nestedRoot = getNestedRoot(root, nestedKey)
         val criteriaKey = nestedKey[nestedKey.size - 1]
-
-        when (criteria.operation) {
-            SearchOperation.EQUALS -> return builder.equal(nestedRoot.get<String>(criteriaKey), criteria.value)
-            SearchOperation.NOT_EQUALS -> return builder.notEqual(nestedRoot.get<String>(criteriaKey), criteria.value)
-            SearchOperation.GREATER_THAN -> return builder.greaterThan(nestedRoot.get<Double>(criteriaKey), criteria.value!!.toDouble())
-            SearchOperation.LESS_THAN -> return builder.lessThan(nestedRoot.get<Double>(criteriaKey), criteria.value!!.toDouble())
-            SearchOperation.STARTS_WITH -> return builder.like(nestedRoot.get(criteriaKey), criteria.value + "%")
-            SearchOperation.ENDS_WITH -> return builder.like(nestedRoot.get(criteriaKey), "%" + criteria.value)
-            SearchOperation.CONTAINS -> return builder.like(builder.lower(nestedRoot.get<String>(criteriaKey).`as`(String::class.java)), "%" + criteria.value + "%")
```

Comment:
So it seems the contains operator was comparing in ignored case in the old method; however with the changes proposed here, this functionality got removed. My PR #10 was quick solution to fix the problem, however with the recent changes it is outdated. Is the intention still to keep the ignored casing here -- or move away from this?

Reply:
Indeed, we want to have case sensitive lookups by default. However, we want to implement a parameter to the annotation to permit case insensitive lookups for cases like yours. The goal is to be able to set case sensitivity by fields independently.